TCD1304.py

```diff
@@ -54,36 +57,36 @@ def __init__(
         self.min_voltagerange_clock = min_voltagerange_clock
         self.max_voltagerange_clock = max_voltagerange_clock
         self.pwmgen = PWMGenerator() #once assigned in __init__ self.pwmgen can be used in other methods
+        self.scope = Oscilloscope ()
+        
         
     def power_source ():        # puts a voltage of 4V on PV1
         ps = PowerSupply()
         ps.pv1 = 4
         ps.pv1
     
     def master_clock (self):        # puts PWM with frequency of 2MHz on SQ1; masterclock is the fastest clock
-        prescaler = int(math.log(oscillator_frequency / frequency_master_clock) / math.log(2))   # When setting a frequency by mapping the reference clock directly to a PWM output, only frequencies which are even factors of 128 MHz (the frequency of the PSLab's main oscillator) are available. The frequency is therefore not set by specifying the frequency itself, but by setting a prescaler.
         self.pwmgen.map_reference_clock(["SQ1"], prescaler)    
     
-    def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime
-        return self.pwmgen.generate() #is the return attribute the one to choose in order to set a second clock?
+    def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime. Running the Sensor in shutter mode try with a frequency of 1/4 of master_clock. tint (min) = 10µs
+        self.pwmgen.generate(["SQ2"], prescaler/4 ) 

```

Comment:
`PWMGenerator.generate` takes the actual frequency as an argument, not a prescaler. I think duty cycle must also be provided.